cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -266,6 +266,7 @@ public void create(Id.Stream streamId) throws Exception {
 
   @Override
   public void create(final Id.Stream streamId, @Nullable final Properties props) throws Exception {
+    assertNamespaceHomeExists(streamId.getNamespace());
```

Comment:
Is this needed? I mean can you access ``FileStreamAdmin`` outside of the HTTP Handler, which would make sure that the namespace exists.

It may be good to double check the directory like you are doing, but just wondering.